scripts/builder_setup.sh

```diff
@@ -41,6 +41,21 @@ export GO_VERSION=$(cat $GOPATH/src/github.com/DataDog/datadog-agent/.go-version
 export IBM_MQ_VERSION=9.2.4.0-IBM-MQ-DevToolkit
 #export IBM_MQ_VERSION=9.2.2.0-IBM-MQ-Toolkit
 
+# Helper to run a bash command with retries, with an exponential backoff.
+# Returns 1 if the provided command fails every time, 0 otherwise.
+function do_with_retries() {
+    local command="$1"
+    local retries="$2"
+    local res=0
+
+    for i in $(seq 0 $2); do
+        res=0
+        sleep $((2**$i))
```

Comment:
Is there a specific reason why you `sleep` before launching the command? I'm asking because I'm used to seeing the other way around.

Reply:
If you sleep before, then you add a 1 second wait time for all invocations. If you sleep after, you add a 2^n wait time to failures (except if you make the logic below more complex and have a break if you know you're at the last iteration).

So it's a tradeoff between: increasing code complexity, or waiting a short bit every time, or waiting a long time on failures. I chose the second option here because I don't want to write complex code in bash (that is a recipe for disaster), and a one second wait is negligible compared to the general duration of the job.

Reply:
Do you sleep at the first iteration? Since `$i` starts at `0` first time it would do `sleep 0`, which is smart

```bash
> for i in $(seq 0 3); do echo $i;done  
0
1
2
3
```

Reply:
Yes, you do sleep one second, because we're not doing `sleep $i`, but `sleep $((2**$i))` (to have an exponential backoff), which (for i=0) is `2^0 = 1`.